catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -242,6 +242,30 @@ public String getResponseWrapper(String responseType, Object response) {
                 EndpointUtil::firstInWinsMerge));
   }
 
+  public Map<String, Result> getMetacardsByFilter(Filter filter) {
+    ResultIterable resultIterable =
+        resultIterable(
+            catalogFramework,
+            new QueryRequestImpl(
+                new QueryImpl(
+                    filter,
+                    1,
+                    pageSize,
+                    new SortByImpl(Core.MODIFIED, SortOrder.DESCENDING),
+                    false,
+                    TimeUnit.SECONDS.toMillis(10)),
```

Comment:
Shouldn't we use timeouts that are already configured in various places of the application?

Reply:
This is something we can revisit when we start cleaning up `EndpointUtil` since it has essentially become a dumping ground for shared code. But as of right now, any query leveraging this and other query utility functions **do not** pull any timeout configuration. These values are all hardcoded.